File: .archive/temp_tests/check_chapters.go
package main
import (
	"context"
	"fmt"
	"time"
	"go.mongodb.org/mongo-driver/bson"
	"go.mongodb.org/mongo-driver/bson/primitive"
	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
)
func main() {
	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
	defer cancel()
	client, _ := mongo.Connect(ctx, options.Client().ApplyURI("mongodb://localhost:27017"))
	defer client.Disconnect(ctx)
	db := client.Database("qingyu_read")
	bookID, _ := primitive.ObjectIDFromHex("696f35c4cee9d6ed15e66935")
	// 检查章节
	count, _ := db.Collection("chapters").CountDocuments(ctx, bson.M{"book_id": bookID})
	fmt.Printf("chapters collection count with book_id %s: %d\n", bookID.Hex(), count)
	// 查看前5个章节
	cursor, _ := db.Collection("chapters").Find(ctx, bson.M{"book_id": bookID}, options.Find().SetLimit(5))
	defer cursor.Close(ctx)
	var results []bson.M
	if err := cursor.All(ctx, &results); err == nil {
		for i, r := range results {
			fmt.Printf("Chapter %d: %+v\n", i+1, r)
		}
	}
}

File: cmd/seeder/check_chapters.js
// MongoDB 查询脚本
var books = db.books.find({title: /修仙/}, {_id: 1, title: 1}).toArray();
print("书籍列表:");
books.forEach(book => {
  var chapterCount = db.chapters.countDocuments({book_id: book._id});
  print("  书籍: " + book.title + ", ID: " + book._id + ", 章节数: " + chapterCount);
  
  if (chapterCount > 0) {
    var chapters = db.chapters.find({book_id: book._id}).limit(1).toArray();
    print("    第一章节ID: " + chapters[0]._id);
  }
});
